source/gui/guiHelper.py

```diff
@@ -168,9 +214,16 @@ def __init__(self, parent, labelText, wxCtrlClass, **kwargs):
 			@param kwargs: The keyword arguments used to instantiate the wxCtrlClass
 		"""
 		object.__init__(self)
-		self._label = wx.StaticText(parent, label=labelText)
-		self._ctrl = wxCtrlClass(parent, **kwargs)
-		self._sizer = associateElements(self._label, self._ctrl)
+		# Dynamically create a class inheriting from BaseLabeledControl and wxCtrlClass.
+		# This class acts as wxCtrlClass but overloads the Enable and Disable methods
+		# in order to also enable or disable self._label at the same time as the control.
+		LabeledControl = type(
+			f'Labeled{wxCtrlClass.__name__}',
+			(BaseLabeledControl, wxCtrlClass),
+			{},
+		)
+		self._ctrl = LabeledControl(parent, labelText=labelText, wxCtrlClass=wxCtrlClass, **kwargs)
+		self._sizer = associateElements(self._ctrl.label, self._ctrl)
 

```

Comment:
Instead you might do something like the following in class `LabeledControlHelper`. Note, I haven't tested the following code. Though it should get you most of the way there. Essentially, at runtime it creates a new subclass of the wx control, with `Enable` and `Disable` overridden. These then trigger a custom event before calling their parent implementations. A listener is bound to this event which will enable / disable the label.

In the past I've run into issues with wx trying to call handlers on destroyed objects, so we unbind if the label is destroyed.

``` python
	import wx.lib.newevent
	# When the control is enabled / disabled this event is raised.
	# A handler is automatically added to the control to ensure the label is also enabled/disabled.
	EnableChanged, EVT_ENABLE_CHANGED = wx.lib.newevent.NewEvent()

	def __init__(self, parent: wx.Window, labelText: str, wxCtrlClass: wx.Control, **kwargs):
		""" @param parent: An instance of the parent wx window. EG wx.Dialog
			@param labelText: The text to associate with a wx control.
			@param wxCtrlClass: The class to associate with the label, eg: wx.TextCtrl
			@param kwargs: The keyword arguments used to instantiate the wxCtrlClass
		"""
		object.__init__(self)
		self._label = _label = wx.StaticText(parent, label=labelText)

		class wxCtrlWithEnableEvnt(wxCtrlClass):
			def Enable(self, enable=True):
				evt = LabeledControlHelper.EnableChanged(isEnabled=enable)
				wx.PostEvent(self, evt)
				super().Enable(enable)

			def Disable(self):
				evt = LabeledControlHelper.EnableChanged(isEnabled=False)
				wx.PostEvent(self, evt)
				super().Disable()

		def _onEnableChanged(evt: wx.Event):
			_label.Enable(evt.isEnabled)

		_ctrl: wx.Control = wxCtrlWithEnableEvnt(parent, **kwargs)
		self._ctrl = _ctrl
		_ctrl.Bind(LabeledControlHelper.EVT_ENABLE_CHANGED, _onEnableChanged)
		# prevent _ctrl trying to call events on destroyed objects
		self._label.Bind(
			wx.wxEVT_DESTROY,
			lambda evt: _ctrl.Unbind(
				LabeledControlHelper.EVT_ENABLE_CHANGED,
				handler=_onEnableChanged
			)
		)
		self._sizer = associateElements(_label, _ctrl)
```